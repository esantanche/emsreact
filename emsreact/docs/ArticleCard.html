<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ArticleCard</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ArticleCard</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ArticleCard<span class="signature">(classes, title, image, text_content, article_node_id)</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>The card containing an article, or better, just the first part of it.</p>
<p>The article is rendered as pure text with no html.</p>
<p>It's also truncated and ellipses are added at the truncation point.</p>
<p>Props</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="ArticleCard"><span class="type-signature"></span>new ArticleCard<span class="signature">(classes, title, image, text_content, article_node_id)</span><span class="type-signature"></span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>classes</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>Classes injected by withStyles function</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>title</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Title of the article</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>image</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Relative url of image as stored in Drupal file system</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>text_content</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Full text of article including html tags</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>article_node_id</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last"><p>Node id of article to be used to retrieve
the article from the backend</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_library_cards_ArticleCard.js.html">components_library/cards/ArticleCard.js</a>, <a href="components_library_cards_ArticleCard.js.html#line74">line 74</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="produce_jsx_fragment_for_image_and_text_according_to_width"><span class="type-signature"></span>produce_jsx_fragment_for_image_and_text_according_to_width<span class="signature">(width, classes, image, text_content)</span><span class="type-signature"> &rarr; {jsx}</span></h4>
    

    



<div class="description">
    <p>This function produces the jsx fragment that displays the image and the text of the article.</p>
<p>It put the image above the text for small devices and the image on the left and the text on the right
for large devices.</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>width</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Material UI width (@see https://material-ui.com/layout/breakpoints/#withwidth-)</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>classes</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last"><p>Classes to be used to make the image responsive</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>image</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Relative url of image as stored in Drupal file system</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>text_content</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Full text of article including html tags</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_library_cards_ArticleCard.js.html">components_library/cards/ArticleCard.js</a>, <a href="components_library_cards_ArticleCard.js.html#line96">line 96</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Jsx fragment showing the image and the text of the article</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">jsx</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AlignedGridPane.html">AlignedGridPane</a></li><li><a href="App.html">App</a></li><li><a href="ArticleCard.html">ArticleCard</a></li><li><a href="ArticleScreen.html">ArticleScreen</a></li><li><a href="ArticleService.html">ArticleService</a></li><li><a href="BodyPane.html">BodyPane</a></li><li><a href="BottomGridPane.html">BottomGridPane</a></li><li><a href="ColouredCard.html">ColouredCard</a></li><li><a href="FooterScreen.html">FooterScreen</a></li><li><a href="FrontGridPane.html">FrontGridPane</a></li><li><a href="HeaderMenu.html">HeaderMenu</a></li><li><a href="HeaderScreen.html">HeaderScreen</a></li><li><a href="HomeButtonPane.html">HomeButtonPane</a></li><li><a href="HomeButtonText.html">HomeButtonText</a></li><li><a href="HomeScreen.html">HomeScreen</a></li><li><a href="InnerGridPane.html">InnerGridPane</a></li><li><a href="ListOfArticlesScreen.html">ListOfArticlesScreen</a></li><li><a href="MasterAppBar.html">MasterAppBar</a></li><li><a href="MasterAvatar.html">MasterAvatar</a></li><li><a href="MenuButton.html">MenuButton</a></li><li><a href="MessageDialog.html">MessageDialog</a></li><li><a href="PickMyBrainDialog.html">PickMyBrainDialog</a></li><li><a href="PickMyBrainMessageService.html">PickMyBrainMessageService</a></li><li><a href="PickMyBrainScreen.html">PickMyBrainScreen</a></li><li><a href="SeparatorPane.html">SeparatorPane</a></li><li><a href="SocialCard.html">SocialCard</a></li><li><a href="SocialScreen.html">SocialScreen</a></li><li><a href="TopicScreen.html">TopicScreen</a></li><li><a href="TopicService.html">TopicService</a></li></ul><h3>Global</h3><ul><li><a href="global.html#theme">theme</a></li><li><a href="global.html#title_to_slug">title_to_slug</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Wed Aug 29 2018 11:52:02 GMT+0000 (UTC)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>