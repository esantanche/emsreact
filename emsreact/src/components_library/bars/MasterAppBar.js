import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import AppBar from '@material-ui/core/AppBar';

// Material UI uses withStyles to inject an array of styles into the DOM as CSS, using JSS
import { withStyles } from "@material-ui/core/styles/index";

// This array of style gets injected into the component as "classes"
const styles = {
    MasterAppBar: {
        borderRadius: "4px",
    }
};

/**
 * The AppBar in the header. Its children are the home button and the menu button.
 *
 * @param {object} classes Classes injected by withStyles function
 */
class MasterAppBar extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {

        const { classes } = this.props;

        return (

            <AppBar position="sticky" className={classes.MasterAppBar}>

                {this.props.children}

            </AppBar>

        )
    }

}

export default withStyles(styles)(MasterAppBar);

