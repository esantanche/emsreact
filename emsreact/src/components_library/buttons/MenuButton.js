import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Button from "@material-ui/core/Button";

// Material UI icons
import MenuIcon from "@material-ui/icons/Menu";

// Material UI functions that make classes available
import { withStyles } from "@material-ui/core/styles/index";

const styles = {
    button: {
        float: "right"
    }
};

/**
 * This is the menu button, the one made of three horizontal lines
 *
 * Props
 * @param {function} onclick Function to call when the user clicks on the menu button
 */
class MenuButton extends Component {

    static propTypes = {
        onclick: PropTypes.func
    };

    render() {

        const { onclick, classes } = this.props;

        return (

            <Button onClick={onclick}
                    className={classes.button}>
                <MenuIcon />
            </Button>

        );
    }
}

export default withStyles(styles)(MenuButton);