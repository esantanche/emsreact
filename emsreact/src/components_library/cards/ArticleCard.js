// Just a test to refactor
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI functions that make classes and width available
import { withStyles } from "@material-ui/core/styles/index";
import withWidth from '@material-ui/core/withWidth';

// Material UI components
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import CardActions from '@material-ui/core/CardActions';
import IconButton from '@material-ui/core/IconButton';

// Material UI icons
import MoreHoriz from '@material-ui/icons/MoreHoriz';

// App configuration
import { APP_CONFIGURATION } from '../../appConfiguration';

// Library used to strip html tags from text
import striptags from 'striptags';

// Used to cut text and add '...'
import Dotdotdot from 'react-dotdotdot';

// From component library
import SeparatorPane from '../panes/SeparatorPane';

// Helper function that takes a title and produces a slug
// which is how we put the title in the url
import { title_to_slug } from '../../helpers/title_to_slug';

const styles = {
    card_header: {
        height: "6em",
        display: "flex",
        alignItems: "center",
        paddingLeft: "24px",
        paddingRight: "24px"
    },
    container: {
        display: "flex",
        flexDirection: "row",
        alignItems: "stretch"
    },
    image_responsive_size: {
        height: "0",
        paddingTop: '100%'
    }
};

class ArticleCard extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired,
        title: PropTypes.string,
        image: PropTypes.string,
        text_content: PropTypes.string,
        article_node_id: PropTypes.number
    };

    produce_jsx_fragment_for_image_and_text_according_to_width(width, classes, image, text_content) {

        const image_fixed_size_according_to_width = {
            sm: "150px",
            md: "150px",
            lg: "180px",
            xl: "200px"
        };

        // We are going to remove any html tag from the text of the article
        // because we want no html tags when showing the text in the article
        // card
        const stripped_text_content = striptags(text_content);

        // When fetching images' url from the Drupal backend, we get relative paths
        // and we have to prepend the backend root url to access the images
        // The module Pathologic in Drupal makes urls absolutes, but not in this case

        if (width === 'xs') {

            return (
                <React.Fragment>

                    <Grid item xs={12}>

                        <CardMedia
                            className={classes.image_responsive_size}
                            image={APP_CONFIGURATION.backendUrl + image}
                            title=""
                        />

                    </Grid>

                    <SeparatorPane/>

                    <Grid item xs={12}>

                        <Typography variant="body2" component="div">

                            <Dotdotdot clamp={5}>

                                {stripped_text_content}

                            </Dotdotdot>

                        </Typography>

                    </Grid>

                </React.Fragment>
            );

        } else {

            // In this case we want the text to be as high as the image.
            // The dotdotdot component will cut the text and add '...'
            // at the end when the text is as high as the image

            return (
                <React.Fragment>

                    <CardMedia
                        style={{ height: image_fixed_size_according_to_width[width],
                                 width: image_fixed_size_according_to_width[width] }}
                        // className={classes.image_fixed_size}
                        image={APP_CONFIGURATION.backendUrl + image}
                        title=""
                    />

                    <Typography variant="body2" component="div"
                                style={{ flex: "1", height: image_fixed_size_according_to_width[width], marginLeft: "10px" }}>

                        <Dotdotdot clamp={image_fixed_size_according_to_width[width]}
                                   style={{ height: image_fixed_size_according_to_width[width] }}>

                            {stripped_text_content}

                        </Dotdotdot>

                    </Typography>

                </React.Fragment>
            );

        }

    }

    render() {

        const title_variant_according_to_width = {
            xs: "title",
            sm: "title",
            md: "subheading",
            lg: "title",
            xl: "title"
        };

        const { classes, title, width, image, text_content, article_node_id } = this.props;

        const jsx_fragment_for_image_and_text_according_to_width =
            this.produce_jsx_fragment_for_image_and_text_according_to_width(width, classes, image, text_content);

        // FIXME link to article to fix

        const slug_for_link_to_article = title_to_slug(title);

        return (

            <Card>

                {/*Used a div because the component CardHeader uses a headline variant of Typography while I    */}
                {/*need the possibility to change font size*/}
                {/*style={{ height: "6em", display: "flex", alignItems: "center", paddingLeft: "24px", paddingRight: "24px" }}*/}

                <div className={classes.card_header}>

                    <Typography variant={title_variant_according_to_width[width]}>
                        {title}
                    </Typography>

                </div>

                <CardContent>

                    <Grid container spacing={0} className={classes.container}>

                        {jsx_fragment_for_image_and_text_according_to_width}

                    </Grid>

                </CardContent>

                <CardActions>
                    <IconButton style={{ marginLeft: "auto" }} href={"/article/" + article_node_id + "/" + slug_for_link_to_article}>
                        <MoreHoriz/>
                    </IconButton>
                </CardActions>

            </Card>

        )

    }

}

export default withWidth()(withStyles(styles)(ArticleCard));
