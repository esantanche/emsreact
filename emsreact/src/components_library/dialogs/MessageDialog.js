import React, { Component } from 'react';
import PropTypes from 'prop-types'

// Material UI components
import Button from "@material-ui/core/es/Button/Button";
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

// Material UI icons
import ErrorOutline from '@material-ui/icons/ErrorOutline';

// On small devices, we want the dialog to go full screen
// This function does the trick
import withMobileDialog from '@material-ui/core/withMobileDialog';

// Material UI function that make classes available
import {withStyles} from "@material-ui/core/styles";

const styles = {
    dialog: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center"
    },
    dialog_title: {
        textAlign: "center"
    },
    error_outline: {
        fontSize: "200%"
    }
};

/**
 * Message dialog to notify successful completion of operations.
 * Used as well for errors. *
 *
 * Props
 * @param {boolean} open True is the dialog should be visible
 * @param {boolean} it_is_an_error_message True if the dialog contains an error message
 * @param {string} title Dialog title
 * @param {string} message Message to display
 * @param {function} onclick Function to call when the user click on OK
 * @param {boolean} fullScreen True if the dialog has to go full screen. This prop
 * is provided by the function withMobileDialog. It will make the dialog full screen
 * on small devices.
 */
class MessageDialog extends Component {

    static propTypes = {
        open: PropTypes.bool.isRequired,
        it_is_an_error_message: PropTypes.bool,
        title: PropTypes.string,
        message: PropTypes.string,
        onclick: PropTypes.func,
        fullScreen: PropTypes.bool.isRequired
    };

    render() {

        const { open, it_is_an_error_message, title, message, onclick, fullScreen, classes } = this.props;

        // Only if it's an error message we display the ErrorOutline icon

        return (

            <Dialog
                open={open}
                onClose={this.handleClose}
                fullScreen={fullScreen}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
                className={classes.dialog}>
                { it_is_an_error_message ?
                    (
                        <DialogTitle className={classes.dialog_title}>
                            <ErrorOutline className={classes.error_outline} />
                        </DialogTitle>
                    ) : ( <React.Fragment></React.Fragment> )
                }
                <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
                <DialogContent>
                    <DialogContentText id="alert-dialog-description">
                        {message}
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button onClick={onclick} color="primary" autoFocus>
                        OK
                    </Button>
                </DialogActions>
            </Dialog>

        )

    }

}

export default withMobileDialog()(withStyles(styles)(MessageDialog));
