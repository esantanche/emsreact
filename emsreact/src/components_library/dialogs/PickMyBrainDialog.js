import React, { Component } from 'react';

// Material UI components
import Button from "@material-ui/core/es/Button/Button";

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import withMobileDialog from '@material-ui/core/withMobileDialog';

import PropTypes from 'prop-types'

import ErrorOutline from '@material-ui/icons/ErrorOutline';
import AppBar from "@material-ui/core/AppBar/AppBar";
import Toolbar from "@material-ui/core/Toolbar/Toolbar";
import IconButton from "@material-ui/core/IconButton/IconButton";
import PickMyBrainScreen from "../../pick_my_brain/PickMyBrainScreen";
import CardContent from "@material-ui/core/CardContent";
import {withStyles} from "@material-ui/core/styles";

import CloseIcon from '@material-ui/icons/Close';
import CardHeader from "@material-ui/core/es/CardHeader/CardHeader";
import ColouredCard from "../cards/ColouredCard";


// ./src/components_library/dialogs/PickMyBrainDialog.js
// Line 73:  'classes' is not defined    no-undef
// Line 76:  'CloseIcon' is not defined  react/jsx-no-undef
//
// Search for the keywords to learn more about each error.

// Is position relative actually needed?
const styles = {
    appBar: {
        position: 'relative',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'flex-end'
    }
};

class PickMyBrainDialog extends Component {

    // FIXME needed?
    // static propTypes = {
    //     open: PropTypes.bool.isRequired,
    //     it_is_an_error_message: PropTypes.bool,
    //     title: PropTypes.string,
    //     message: PropTypes.string,
    //     onclick: PropTypes.func,
    //     fullScreen: PropTypes.bool.isRequired
    // };

    static propTypes = {
        open: PropTypes.bool.isRequired,
        onclick: PropTypes.func
    };

    constructor(props) {
        super(props);

        this.state = {
            // The element the menu is anchored to, typically a button that opens the menu
            // When anchorElement is null, the menu is invisible
            // FIXME ? ok?
            open: false
        };

    }

    // handleClose = () => {
    //     this.setState({ open: false });
    // };

    render() {

        // FIXME fix all these props

        const { open, it_is_an_error_message, title, message, onclick, fullScreen, classes } = this.props;

        // style={{ display: "flex", flexDirection: "column", alignItems: "center" }}
        // style={{ display: "flex", flexDirection: "row" }}

        return (

            <Dialog
                open={open}
                onClose={this.handleClose}
                fullScreen={fullScreen}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
                >
                <AppBar className={classes.appBar}>
                    <Toolbar>
                        <IconButton color="inherit" onClick={onclick} aria-label="Close">
                            <CloseIcon />
                        </IconButton>
                    </Toolbar>
                </AppBar>

                <ColouredCard light={true}>

                    <CardHeader title="Pick My Brain!"
                                subheader="Or just ask me a question" />

                    <CardContent>

                        <PickMyBrainScreen onclick={onclick} />

                    </CardContent>

                </ColouredCard>

            </Dialog>

        )

    }

}

export default withMobileDialog()(withStyles(styles)(PickMyBrainDialog));


// export default withMobileDialog()(PickMyBrainDialog);



