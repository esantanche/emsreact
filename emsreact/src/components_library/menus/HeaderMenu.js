import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import Menu from "@material-ui/core/Menu/Menu";
import MenuItem from "@material-ui/core/MenuItem/MenuItem";

/**
 * Header menu.
 *
 * Props
 * @param {object} anchorElement The element this menu should anchor to.
 * It gives this menu a position.
 * @param {function} handleMenuClose Function to call when the user clicks
 * on a menu item
 */
class HeaderMenu extends Component {

    static propTypes = {
        anchorElement: PropTypes.object,
        handleMenuClose: PropTypes.func
    };

    render() {

        const { anchorElement, handleMenuClose } = this.props;

        // To close this menu, just make anchorElement null

        // Pick My Brain menu item
        // It calls handleMenuClose with no url and true as second parameter
        // This means that the parent component will open the Pick My Brain Dialog
        // See HeaderScreen.js and PickMyBrainDialog.js

        return (

            <Menu
                id="header-menu"
                anchorEl={anchorElement}
                open={Boolean(anchorElement)}
                onClose={handleMenuClose}>

                <MenuItem onClick={() => handleMenuClose("/")}>Home</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/how-i-work")}>How I work</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/the-tools-i-use")}>The tools I use</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/success-stories")}>Success stories</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/testimonials")}>Testimonials</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/talking-about-my-experiences")}>Articles</MenuItem>
                <MenuItem onClick={() => handleMenuClose("/articles/where-how")}>Where? How?</MenuItem>
                <MenuItem onClick={() => handleMenuClose("", true)}>
                    Pick My Brain!
                </MenuItem>

            </Menu>

        )

    }

}

export default HeaderMenu;


