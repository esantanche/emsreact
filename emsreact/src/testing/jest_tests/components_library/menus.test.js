// FIXME docs

// import ReactDOM from "react-dom";
import React from "react";
import renderer from 'react-test-renderer';
//import MasterAvatar from "../../../components_library/avatars/MasterAvatar";

import red from "@material-ui/core/colors/red";
import MasterAppBar from "../../../components_library/bars/MasterAppBar";
import MenuButton from "../../../components_library/buttons/MenuButton";

import { shallow } from 'enzyme';
import MessageDialog from "../../../components_library/dialogs/MessageDialog";
import ArticleCard from "../../../components_library/cards/ArticleCard";

import { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import PickMyBrainDialog from "../../../components_library/dialogs/PickMyBrainDialog";
import HeaderMenu from "../../../components_library/menus/HeaderMenu";
import ReactDOM from "react-dom";

// FIXME evertyhing

// FIXME Here we test components in components_library/avatars

{/*<MessageDialog open={this.state.error_dialog_is_open}*/}
               {/*it_is_an_error_message={true}*/}
               {/*title="Something went wrong"*/}
               {/*message="Please, save your message, reload the page and try again."*/}
               {/*onclick={this.handleOnClickErrorDialog.bind(this)}/>*/}

{/*<MessageDialog open={this.state.message_dialog_is_open}*/}
{/*it_is_an_error_message={false}*/}
{/*title="Message sent"*/}
{/*message="Thank you for your message!"*/}
{/*onclick={this.handleOnClickDialog.bind(this)}/>*/}

{/*<MessageDialog open={this.state.message_dialog_is_open}*/}
               {/*it_is_an_error_message={false}*/}
               {/*title="Message sent"*/}
               {/*message="Thank you for your message!"*/}
               {/*onclick={this.handleOnClickDialog.bind(this)}/>*/}


//MessageDialog

// FIXME

describe('Testing HeaderMenu', function() {

    // FIXME

    {/*<HeaderMenu anchorElement={this.state.anchorElement}*/}
                {/*handleMenuClose={this.handleMenuClose.bind(this)}/>*/}

    const mockHandle = jest.fn();

    const div = document.createElement('div');

    const header_menu_to_be_tested = ( <HeaderMenu anchorElement={div}
                                                   handleMenuClose={mockHandle}/> );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('HeaderMenu renders correctly', () => {

        const tree = shallow(header_menu_to_be_tested);

        expect(EnzymeToJson(tree)).toMatchSnapshot();
        // const tree = renderer
        //     .create(header_menu_to_be_tested)
        //     .toJSON();
        // expect(tree).toMatchSnapshot();



        // const tree = renderer
        //     .create(message_dialog_error_message_to_be_tested)
        //     .toJSON();
        // expect(tree).toMatchSnapshot();
    });

//
// // FIXME
//     it('MessageDialog renders correctly, confirmation message', () => {
//
//         const tree = shallow(message_dialog_confirmation_message_to_be_tested);
//
//         expect(EnzymeToJson(tree)).toMatchSnapshot();
//
//         // const tree = renderer
//         //     .create(message_dialog_confirmation_message_to_be_tested)
//         //     .toJSON();
//         // expect(tree).toMatchSnapshot();
//     });
//
//     it('MessageDialog, test click event', () => {
//         const mockCallBack = jest.fn();
//
//         const dialog = mount(<MessageDialog open={true}
//                                             it_is_an_error_message={true}
//                                             title="Something went wrong"
//                                             message="Please, save your message, reload the page and try again."
//                                             onclick={mockCallBack}/>);
//
//         dialog.find('button').simulate('click');
//         expect(mockCallBack.mock.calls.length).toEqual(1);
//     });

});
//
// // FIXME
//
// describe('Testing PickMyBrainDialog', function() {
//
//     {/*<PickMyBrainDialog open={this.state.pick_my_brain_dialog_is_open}*/}
//     {/*onclick={this.handleOnClickPickMyBrainDialog.bind(this)}/>*/}
//
//     const pick_my_brain_dialog_to_be_tested = ( <PickMyBrainDialog open={true} /> );
//
//
//     it('PickMyBrainDialog renders correctly', () => {
//
//         const tree = shallow(pick_my_brain_dialog_to_be_tested);
//
//         expect(EnzymeToJson(tree)).toMatchSnapshot();
//
//     });
//
//     // input.simulate('change', { target: { value: 'Hello' } })
//
//     // FIXME want to test the click, but it cannot work if the fields are empty
//     // it('PickMyBrainDialog, test click event', () => {
//     //     const mockCallBack = jest.fn();
//     //
//     //     const dialog = shallow(pick_my_brain_dialog_to_be_tested);
//     //
//     //     dialog.find('input').simulate('change', { target: { value: 'esantanche@gmail.com' }});
//     //
//     //     expect(mockCallBack.mock.calls.length).toEqual(1);
//     // });
//
// });