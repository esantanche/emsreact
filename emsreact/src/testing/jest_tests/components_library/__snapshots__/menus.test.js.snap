// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HeaderMenu HeaderMenu renders correctly 1`] = `
<WithStyles(Menu)
  anchorEl={<div />}
  id="header-menu"
  onClose={[Function]}
  open={true}
>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    Home
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    How I work
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    The tools I use
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    Success stories
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    Testimonials
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    Articles
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    Where? How?
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    onClick={[Function]}
  >
    Pick My Brain!
  </WithStyles(MenuItem)>
</WithStyles(Menu)>
`;
